pypiserver/_app.py

```diff
@@ -389,12 +391,29 @@ def json_info(project):
     if not packages:
         raise HTTPError(404, f"package {project} not found")
 
+    package_links = defaultdict(list)
+    for pkg in packages:
+        package_links[pkg.version].append(pkg.relfn_unix)
+    # links = [pkg.relfn_unix for pkg in packages]
+
     latest_version = packages[0].version
     releases = {}
     req_url = request.url
-    for x in packages:
-        releases[x.version] = [
-            {"url": urljoin(req_url, "../../packages/" + x.relfn)}
+    for package in packages:
+        matching_links = []
+        for version, links in package_links.items():
+            if version == package.version:
+                matching_links += links
```

Comment:
It's sure a nit, but what do you think about doing it this way? 😀 

```suggestion
        matching_links = [links for version, links in package_links.items() if version == package.version]
```